sphinx_js/typedoc.py

```diff
@@ -194,7 +194,15 @@ def _convert_node(self, node) -> Tuple[TopLevel, List[dict]]:
             # many attr of Functions.
             sigs = node.get('signatures')
             first_sig = sigs[0]  # Should always have at least one
-            first_sig['sources'] = node['sources']
+            def rec_helper(node):
+                if "sources" in node["__parent"]:
+                    return node["__parent"]["sources"]
+                else:
+                    rec_helper(node["__parent"])
```

Comment:
I know it is styling, but how about

```suggestion
            def rec_helper(node):
                if "sources" not in node["__parent"]:
                    rec_helper(node["__parent"])
                
                return node["__parent"]["sources"]
```

Reply:
Come to think of it, there's an apparent bug in this code to start with. The recursive handling isn't returning at all-- this is getting ignored.

I pulled this from someone else's work. I'll modify it and see if anything changes.